src/context/auth/authContextController/AuthContextController.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+import { AuthContext } from '../authContext/AuthContext';
+import { AuthContextControllerProps } from './AuthContextController.types';
+import { createCheckIsLoggedInAction } from 'api/actions/auth/authActions';
+import { useQuery } from 'api/hooks';
+import { AuthStorage } from '../authStorage.enum';
+
+export const AuthContextController = ({ children }: AuthContextControllerProps) => {
+  const [token, setToken] = useState<string | null | undefined>(localStorage.getItem(AuthStorage.TOKEN));
+
+  const { error } = useQuery<string>(createCheckIsLoggedInAction());
+
+  useEffect(() => {
+    if (error) {
+      setToken(null);
+    }
+  }, [error]);
+
+  useEffect(() => {
+    token ? localStorage.setItem(AuthStorage.TOKEN, token) : localStorage.setItem(AuthStorage.TOKEN, '');
```

Comment:
Dwie rzeczy:
1. Jak można skrócić ten zapis? Bo z tego co widzę, że dwa razy robi się to samo
2. Czy przy inicjalizacji token w local storagu nadpisuje się tak czy tak?